tests/playwright/src/model/pages/preferences-page.ts

```diff
@@ -23,16 +23,62 @@ import { SettingsPage } from './settings-page';
 
 export class PreferencesPage extends SettingsPage {
   readonly heading: Locator;
+  readonly searchbar: Locator;
   readonly kubePathInput: Locator;
 
   constructor(page: Page) {
     super(page, 'Preferences');
     this.heading = this.header.getByLabel('Title', { exact: true });
+    this.searchbar = this.header.getByLabel('search preferences');
     this.kubePathInput = this.content.getByLabel(
       'Path to the Kubeconfig file for accessing clusters. (Default is usually ~/.kube/config)',
     );
   }
 
+  getPreferenceRowByName(name: string): Locator {
+    const preferenceLocator = this.content.getByText(name, { exact: true }).last();
+    return preferenceLocator.locator('xpath=ancestor::div[@class="flex flex-row justify-between"]');
+  }
+
+  async isPreferenceManaged(name: string): Promise<boolean> {
+    const preferenceRow = this.getPreferenceRowByName(name);
+    const managedIndicator = await preferenceRow.getByText('Managed').all();
+    return managedIndicator.length > 0;
+  }
+
+  async resetPreference(name: string): Promise<void> {
+    const preferenceRow = this.getPreferenceRowByName(name);
+    const resetButton = preferenceRow.getByRole('button', { name: 'Reset to default value' });
+    await playExpect(resetButton).toBeVisible();
+    await resetButton.click();
+  }
+
+  async getAppearancePreferenceValue(): Promise<string> {
+    const appearancePreferenceRow = this.getPreferenceRowByName('Appearance');
+    await playExpect(appearancePreferenceRow).toBeAttached();
+    await appearancePreferenceRow.scrollIntoViewIfNeeded();
+    await playExpect(appearancePreferenceRow).toBeVisible();
+
+    const preferenceInput = appearancePreferenceRow.getByLabel('hidden input');
+    await playExpect(preferenceInput).toBeAttached();
+    return await preferenceInput.inputValue();
+  }
+
+  async setAppearancePreference(value: string): Promise<void> {
+    const appearancePreferenceRow = this.getPreferenceRowByName('Appearance');
+    await playExpect(appearancePreferenceRow).toBeAttached();
+    await appearancePreferenceRow.scrollIntoViewIfNeeded();
+    await playExpect(appearancePreferenceRow).toBeVisible();
+
+    const selectionButton = appearancePreferenceRow.locator('button#input-standard-preferences\\.appearance');
```

Comment:
```suggestion
const selectionButton = appearancePreferenceRow.getByLabel('Select between light or dark mode, or use your system setting.');
```

The outer div has aria-label field, does that not work?

Reply:
if it doesn't work like this than it should work to select the other div with it and then use getByRole('button') on the outer div locator to get the button itself, as an alternative method.

Reply:
I will try

Reply:
surprisingly clicking on the div itself works

Reply:
In the most cases the action elements have `aria-label` with such description. Sometimes it might be missing, in for example minikube extension, but that should be added there rather than updating all dom here. So I would go with this approach which is sufficient for the tests. Closing https://github.com/podman-desktop/podman-desktop/pull/15267 as it did not help in this situation.